### PR TITLE
fix --benchmark not working on codeflash itself

### PR DESCRIPTION
**Pull request by KRRT7**

### CODE REVIEW

File: codeflash/benchmarking/instrument_codeflash_trace.py [hunk 1]
```diff
@@ -105,6 +105,17 @@ def add_codeflash_decorator_to_code(code: str, functions_to_optimize: list[Funct
 def instrument_codeflash_trace_decorator(file_to_funcs_to_optimize: dict[Path, list[FunctionToOptimize]]) -> None:
     """Instrument codeflash_trace decorator to functions to optimize."""
     for file_path, functions_to_optimize in file_to_funcs_to_optimize.items():
+        # Skip codeflash's own benchmarking and picklepatch modules to avoid circular imports
+        # (codeflash_trace.py imports from picklepatch, and instrumenting these would cause
+        # them to import codeflash_trace back, creating a circular import)
+        parts = file_path.parts
+        # Find the last "codeflash" in path (handles nested paths like .../codeflash/codeflash/...)
+        codeflash_indices = [i for i, p in enumerate(parts) if p == "codeflash"]
```
Comment (aseembits93) [inline codeflash/benchmarking/instrument_codeflash_trace.py]: wonder if the .partition optimization you found for pypa packaging could apply here @KRRT7

File: codeflash/code_utils/concolic_utils.py [hunk 2]
```diff
@@ -33,9 +33,9 @@ def _transform_assert_line(self, line: str) -> Optional[str]:
             indent, assert_method, args = unittest_match.groups()
 
             if args:
-                arg_parts = self._split_top_level_args(args)
-                if arg_parts and arg_parts[0]:
-                    return f"{indent}{arg_parts[0]}"
+                arg_parts = self._first_top_level_arg(args)
```
Comment (aseembits93) [inline codeflash/code_utils/concolic_utils.py:36]: the older `_split_top_level_args` is not used anywhere else I believe, you can remove it

File: codeflash/code_utils/concolic_utils.py [hunk 3]
```diff
@@ -67,6 +67,17 @@ def __init__(self) -> None:
         self.assert_re = re.compile(r"\s*assert\s+(.*?)(?:\s*==\s*.*)?$")
         self.unittest_re = re.compile(r"(\s*)self\.assert([A-Za-z]+)\((.*)\)$")
 
+    def _first_top_level_arg(self, args: str) -> str:
```
Comment (aseembits93) [inline codeflash/code_utils/concolic_utils.py:47]: there could be some edge cases in this function, you should write some unit test cases for this.

File: codeflash/benchmarking/instrument_codeflash_trace.py (hunk 1 above)
Comment (KRRT7) [reply codeflash/benchmarking/instrument_codeflash_trace.py]: good idea, I'll try it out

**Comment by KRRT7** [comment]
@aseembits93 added tests
